src/plugins/data/public/search/expressions/esaggs.ts

```diff
@@ -58,6 +62,7 @@ export function getFunctionDefinition({
         timeFields: args.timeFields,
         timeRange: get(input, 'timeRange', undefined),
         getNow,
+        executionContext: executionContext?.toJSON(),
```

Comment:
The `data` plugin data fetching model is a bit different from the one most plugins use. It doesn't send an HTTP request for every search operation, but batches them according to some internal rules and send them in bulk. In turn, the Kibana server parses the batch and issues a dedicated search request for every search operation. Kibana server streams Elasticsearch server response back to the browser as soon as every search operation is finished.
![Untitled-2021-07-05-1246](https://user-images.githubusercontent.com/3198181/125286449-18d2f500-e31c-11eb-8331-6723ad24a3e3.png)



So I had to refactor the plugin a bit to support `executionContext` propagation.

Reply:
why are we passing json here ?

Reply:
the downstream code requires `SerializableState`, so I had to convert to `json` before calling `fetch$`

Reply:
SerializableState is any object that can be serialized, seems KibanaExecutionContext already is

Reply:
`KibanaExecutionContext` is serializable, but the `handler` receives https://github.com/elastic/kibana/pull/105206/files#diff-357bbbe4bfbe5a09c1e453b99baba2082a43ef48d34d1677831cc3c485f1bca6R70

Reply:
is there a way to make it serializable without converting to json string ?

Reply:
> is there a way to make it serializable without converting to json string ?

It's not a json string, it returns a json object. Actually, any object with [toJSON method is serializable with JSON.stringify](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify), but `SerializableState` interface doesn't support it.